src/arns.lua

```diff
@@ -839,130 +874,59 @@ function arns.assertValidIncreaseUndername(record, qty, currentTimestamp)
-
---- Submits a bid to an auction
---- @param name string The name of the auction
---- @param bidAmount number|nil The amount of the bid
---- @param bidder string The address of the bidder
---- @param timestamp number The timestamp of the bid
---- @param processId string The processId of the bid
---- @param type string The type of the bid
---- @param years number|nil The number of years for the bid
---- @param msgId string The current messageId
---- @param fundFrom string|nil The intended payment sources; one of "any", "balance", or "stakes". Default "balance"
---- @return AuctionBidResult auctionBidResult The result of the bid
-function arns.submitAuctionBid(name, bidAmount, bidder, timestamp, processId, type, years, msgId, fundFrom)
-	fundFrom = fundFrom or "balance"
-	local auction = arns.getAuction(name)
-	assert(auction, "Auction not found")
-	assert(
-		timestamp >= auction.startTimestamp and timestamp <= auction.endTimestamp,
-		"Bid timestamp is outside of auction start and end timestamps"
-	)
-	local requiredBid = auction:getPriceForAuctionAtTimestamp(timestamp, type, years)
-	local floorPrice = auction:floorPrice(type, years) -- useful for analytics, used by getPriceForAuctionAtTimestamp
-	local startPrice = auction:startPrice(type, years) -- useful for analytics, used by getPriceForAuctionAtTimestamp
-	local requiredOrBidAmount = bidAmount or requiredBid
-
-	local finalBidAmount = requiredBid
-
-	-- check if bidder is eligible for ArNS discount
-	if gar.isEligibleForArNSDiscount(bidder) then
-		local discount = math.floor(finalBidAmount * constants.ARNS_DISCOUNT_PERCENTAGE)
-		finalBidAmount = finalBidAmount - discount
-	end
-
-	assert(requiredOrBidAmount >= requiredBid, "Bid amount is less than the required bid of " .. requiredBid)
-
-	-- check the balances of the bidder
-	local fundingPlan = gar.getFundingPlan(bidder, finalBidAmount, fundFrom)
-	assert(fundingPlan and fundingPlan.shortfall == 0 or false, "Insufficient balances")
-
-	-- apply the funding plan
-	local fundingResult = gar.applyFundingPlan(fundingPlan, msgId, timestamp)
-
-	local record = {
-		processId = processId,
-		startTimestamp = timestamp,
-		endTimestamp = type == "lease" and timestamp + constants.oneYearMs * years or nil,
-		undernameLimit = constants.DEFAULT_UNDERNAME_COUNT,
-		purchasePrice = finalBidAmount,
-		type = type,
-	}
+function arns.getReturnedNamePremiumMultiplier(startTimestamp, currentTimestamp)
+	local timestampDiff = currentTimestamp - startTimestamp
+	assert(timestampDiff < constants.returnedNamePeriod, "Timestamp is outside of the returned name period")
+	-- The percentage of the period that has passed e.g: 0.5 if half the period has passed
+	local percentageOfReturnedNamePeriodPassed = timestampDiff / constants.returnedNamePeriod
+	-- Take the inverse so that a fresh returned name has the full multiplier, and a name almost expired has a multiplier close to base price
+	local inversePercentage = 1 - percentageOfReturnedNamePeriodPassed
```

Comment:
Friendlier name might be `pctOfReturnPeriodRemaining`

Reply:
f34ac93